src/services/expensesService.js

```diff
@@ -0,0 +1,54 @@
+let expenses = [];
+
+const resetExpenses = () => {
+  expenses = [];
+};
+
+const getAllExpenses = (userId, categories, from, to) => {
+  return expenses
+    .filter((expense) => !userId || expense.userId === +userId)
+    .filter((expense) => !categories || expense.category === categories)
+    .filter((expense) => !from || new Date(expense.spentAt) >= new Date(from))
+    .filter((expense) => !to || new Date(expense.spentAt) <= new Date(to));
+};
+
+const getExpenseById = (id) => {
+  return expenses.find((expense) => expense.id === +id);
+};
+
+const createExpense = (userId, spentAt, title, amount, category, note) => {
+  const expense = {
+    id: Math.floor(Math.random() * 1000),
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  };
+
+  expenses.push(expense);
+
+  return expense;
+};
+
+const updateExpense = (id, data) => {
+  const expense = getExpenseById(id);
+
+  Object.assign(expense, data);
+
+  return expense;
```

Comment:
In the `updateExpense` function, you are not handling the case where `getExpenseById` might return `undefined` (when there is no expense with the given id). This could lead to a runtime error. You should add a check to ensure `expense` is not `undefined` before trying to update it.